### DISCUSSION TITLE
[postcss-preset-env] what happen when a feature is rejected ?

### DISCUSSION THREAD

**Discussion by fchastanet** [csstools/postcss-plugins#231] [Q&A]
Hello,

Let's say I'm using a stage 2 feature and this feature is rejected. Would this feature be kept in cssdb ?
if this is not the case, my next css build could miss the polyfill and my frontend being impacted.

**Comment by romainmenke** [comment]
Hi!

Any drop in `stage` or in the number of implementations is indeed very tricky.
Users of `postcss-preset-env` expect a given source to have a stable output for a given config.

Normally we would detect these and make it a breaking change in both `cssdb` and `postcss-preset-env`.

This happened with : 
https://github.com/csstools/postcss-color-mod-function

It was only removed when we released `postcss-preset-env` v7 with `PostCSS 8` support.

To help users migrate their source code I created this :

https://github.com/romainmenke/postcss-preset-env-migrations

It has two things and you pick one :

- a tiny plugin pack that includes the removed plugins. (intended as a stop-gap measure)
- a VSCode extension that makes it easy to convert source code to a native alternative. (real solution)

At the moment I intend to keep track of breaking changes in `postcss-preset-env` and add more migrations there.
